File: Project001-SPA/src/app/Models/Hotel.ts
import { Floor } from "./Floor";
import { Room } from "./Room";
import { RoomType } from "./RoomType";

export interface Hotel {
    id: number;
    name: string;
    phone: string;
    location: string;
    rooms: Room[];
    floors: Floor[];
    roomTypes: RoomType[];
}

File: Project001.API/Models/Hotel.cs
using System;
using System.Collections.Generic;

namespace Project001.API.Models{
    public class Hotel{
        public int id{set;get;}
        public string name{set;get;}
        public string phone{set;get;}
        public string location{set;get;}

        public ICollection<Room> rooms{set;get;}
        public ICollection<Floor> floors{set;get;}
        public ICollection<RoomType> roomTypes{set;get;}

    }
}